File: SeriesGuide/src/main/java/com/battlelancer/seriesguide/ui/BaseSearchFragment.java
package com.battlelancer.seriesguide.ui;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.GridView;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import com.battlelancer.seriesguide.R;
import com.uwetrottmann.androidutils.AndroidUtils;
import org.greenrobot.eventbus.EventBus;

public abstract class BaseSearchFragment extends Fragment
        implements AdapterView.OnItemClickListener {

    @BindView(R.id.textViewSearchEmpty) TextView textViewEmpty;
    @BindView(R.id.gridViewSearch) GridView gridView;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_search, container, false);
        ButterKnife.bind(this, view);

        // enable app bar scrolling out of view only on L or higher
        ViewCompat.setNestedScrollingEnabled(gridView, AndroidUtils.isLollipopOrHigher());
        gridView.setOnItemClickListener(this);
        gridView.setEmptyView(textViewEmpty);
        gridView.setFastScrollAlwaysVisible(false);
        gridView.setFastScrollEnabled(true);

        return view;
    }

    @Override
    public void onStart() {
        super.onStart();

        EventBus.getDefault().register(this);
    }

    @Override
    public void onStop() {
        super.onStop();

        EventBus.getDefault().unregister(this);
    }

}

File: app/src/main/java/com/battlelancer/seriesguide/shows/search/BaseSearchFragment.kt
// Copyright 2023 Uwe Trottmann
// SPDX-License-Identifier: Apache-2.0

package com.battlelancer.seriesguide.shows.search

import android.os.Bundle
import android.view.View
import android.widget.GridView
import androidx.core.view.ViewCompat
import androidx.fragment.app.Fragment
import com.battlelancer.seriesguide.util.ThemeUtils
import org.greenrobot.eventbus.EventBus

abstract class BaseSearchFragment : Fragment() {

    abstract val emptyView: View

    abstract val gridView: GridView

    var initialSearchArgs: Bundle? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (savedInstanceState == null) {
            // use initial query (if any)
            val queryEvent = EventBus.getDefault()
                .getStickyEvent(SearchActivityImpl.SearchQueryEvent::class.java)
            if (queryEvent != null) {
                initialSearchArgs = queryEvent.args
            }
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // enable app bar scrolling out of view
        ViewCompat.setNestedScrollingEnabled(gridView, true)
        ThemeUtils.applyBottomPaddingForNavigationBar(gridView)

        emptyView.visibility = View.GONE
    }

    override fun onStart() {
        super.onStart()

        EventBus.getDefault().register(this)
    }

    override fun onStop() {
        super.onStop()

        EventBus.getDefault().unregister(this)
    }

    protected fun updateEmptyState(hasNoResults: Boolean, hasQuery: Boolean) {
        if (hasNoResults && hasQuery) {
            emptyView.visibility = View.VISIBLE
            gridView.visibility = View.GONE
        } else {
            emptyView.visibility = View.GONE
            gridView.visibility = View.VISIBLE
        }
    }

}
